packages/splits-sdk/src/types.ts

```diff
@@ -117,6 +133,184 @@ export type GetSplitBalanceConfig = {
   token?: string
 }
 
+// Warehouse
+export type WarehouseTransferConfig = {
+  receiverAddress: Address
+  tokenAddress: Address
+  amount: bigint
+} & TransactionOverridesDict
+
+export type WarehouseTransferFromConfig = {
+  senderAddress: Address
+  receiverAddress: Address
+  tokenAddress: Address
+  amount: bigint
+} & TransactionOverridesDict
+
+export type WarehouseApproveConfig = {
+  spenderAddress: Address
+  tokenAddress: Address
+  amount: bigint
+} & TransactionOverridesDict
+
+export type WarehouseSetOperatorConfig = {
+  operatorAddress: Address
+  approved: boolean
+} & TransactionOverridesDict
+
+export type WarehouseInvalidateNonceConfig = {
+  nonce: bigint
+} & TransactionOverridesDict
+
+export type WarehouseTemporaryApproveAndCallConfig = {
+  spenderAddress: Address
+  operator: boolean
+  tokenAddress: Address
+  amount: bigint
+  targetAddress: Address
+  data: Hex
+} & TransactionOverridesDict
+
+export type WarehouseTemporaryApproveAndCallBySigConfig = {
+  ownerAddress: Address
+  spenderAddress: Address
+  operator: boolean
+  tokenAddress: Address
+  amount: bigint
+  targetAddress: Address
+  data: Hex
+  nonce: bigint
+  deadline: number
+  signature: Hex
+} & TransactionOverridesDict
+
+export type WarehouseApproveBySig = {
+  spenderAddress: Address
+  operator: boolean
+  tokenAddress: Address
+  amount: bigint
+  nonce: bigint
+  deadline: number
+}
+
+export type WarehouseTemporaryApproveAndCallBySig = {
+  spenderAddress: Address
+  operator: boolean
+  tokenAddress: Address
+  amount: bigint
+  targetAddress: Address
+  data: Hex
+  nonce: bigint
+  deadline: number
+}
+
+export type WarehouseApproveBySigConfig = {
+  ownerAddress: Address
+  spenderAddress: Address
+  operator: boolean
+  tokenAddress: Address
+  amount: bigint
+  nonce: bigint
+  deadline: number
+  signature: Hex
+} & TransactionOverridesDict
+
+export type WarehouseDepositConfig = {
+  receiverAddress: Address
+  tokenAddress: Address
+  amount: bigint
+} & TransactionOverridesDict
+
+export type WarehouseBatchDepositConfig = {
+  receiversAddresses: Address[]
+  tokenAddress: Address
+  amounts: bigint[]
+} & TransactionOverridesDict
+
+export type WarehouseWithdrawConfig = {
+  ownerAddress: Address
+  tokenAddress: Address
+} & TransactionOverridesDict
+
+export type WarehouseBatchWithdrawConfig = {
+  ownerAddress: Address
+  tokensAddresses: Address[]
+  amounts: bigint[]
+  withdrawerAddress: Address
+} & TransactionOverridesDict
+
+export type WarehouseBatchTransferConfig = {
+  receiversAddresses: Address[]
+  tokenAddress: Address
+  amounts: bigint[]
+} & TransactionOverridesDict
+
+export type WarehouseSetWithdrawConfig = {
+  incentivePercent: number
+  paused: boolean
+} & TransactionOverridesDict
+
+export enum SplitV2Type {
+  Push = 'push',
+  Pull = 'pull',
+}
+
+// Split V2
+
+export type SplitV2 = {
```

Comment:
is this like, the type for what gets sent back to the user on getSplitMetadata? if so couple thoughts:

- do we want to combine recipients and allocations together? i think we do that for v1.
- the recipient should be of type `Recipient`, which is an object that has address and optional ens.
- in v1 we're passing back percents for allocation and incentive. basically matches the create split args

i think...if we want to differentiate we can. my preference is keep things consistent though.